templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -385,14 +385,31 @@
           function checkEmail() {
             var data = $("#contact_id", $form).select2('data');
             if (data && data.extra && data.extra.email && data.extra.email.length) {
-              $("#email-receipt", $form).show();
-              $("#email-address", $form).html(data.extra.email);
+              CRM.api4('Email', 'get', {
+                select: ["on_hold"],
+                where: [["contact_id", "=", data.id], ["is_primary", "=", true]]
+              }).then(function(emails) {
+                if (emails[0]['on_hold'] === 0) {
+                  showEmailReceipt();
+                }
+                else {
+                  CRM.alert(ts("No email receipt can be sent as the contact's primary email address is on hold."), ts("Email on hold"));
+                  $("#email-receipt", $form).hide();
+                }
+              }, function(failure) {
+                showEmailReceipt();
+              });
             }
             else {
               $("#email-receipt", $form).hide();
             }
           }
 
+          function showEmailReceipt() {
+            $("#email-receipt", $form).show();
+            $("#email-address", $form).html(data.extra.email);
```

Comment:
data.extra.email is undefined here

On the question of altering the select2, I think you would still want to be able to enter contributions for them, so I don't think you want to prevent them from appearing there. There is maybe the question of whether to also do this alert for contacts with "do not email" set, but that can be a separate topic.

Reply:
Fixed.

Also added a check for do not email contacts as well, I think it makes sense to do so, do not email means do not send emails, so we should not.

Reply:
For altering the select2, I was thinking maybe there was a way to still show those contacts with emails on hold, but not show their email addresses, but I don't think there is.